$OpenBSD: patch-libavcodec_libvpxenc_c,v 1.1 2015/04/28 07:13:15 ajacoutot Exp $

238ec505e263e7d0363798a05c1c28f494a6300a;hp=acb07f33c40c2aaca7b26e5046048ca628643803

libvpxenc: only set noise reduction w/vp8

--- libavcodec/libvpxenc.c.orig	Mon Apr 27 19:48:00 2015
+++ libavcodec/libvpxenc.c	Mon Apr 27 19:49:03 2015
@@ -441,9 +441,10 @@ static av_cold int vpx_init(AVCodecContext *avctx,
         codecctl_int(avctx, VP8E_SET_ARNR_STRENGTH,    ctx->arnr_strength);
     if (ctx->arnr_type >= 0)
         codecctl_int(avctx, VP8E_SET_ARNR_TYPE,        ctx->arnr_type);
-    codecctl_int(avctx, VP8E_SET_NOISE_SENSITIVITY, avctx->noise_reduction);
-    if (avctx->codec_id == AV_CODEC_ID_VP8)
+    if (avctx->codec_id == AV_CODEC_ID_VP8) {
+        codecctl_int(avctx, VP8E_SET_NOISE_SENSITIVITY, avctx->noise_reduction);
         codecctl_int(avctx, VP8E_SET_TOKEN_PARTITIONS,  av_log2(avctx->slices));
+    }
 #if FF_API_MPV_OPT
     FF_DISABLE_DEPRECATION_WARNINGS
     if (avctx->mb_threshold) {
